Problem1.py

```diff
@@ -0,0 +1,35 @@
+# DP-2
+
+## Problem1(https://leetcode.com/problems/paint-house/)
+
+# Approach
+# Initialize the matrix with row as costs and columns as colors. Traverse from bottom to top and set lowest row colors same as given cost matrix.
+# From second last row, check for min cost by adding current cost with cost of column except itself. Eg.: for column 0 check sum current value at column 0 with cost in next row at column 1 & 2. Append min value at that index
+# Similarly traverse till the first row and return the min cost from the 3 cost in column 0, 1 & 2. 
+
+# Time Complexity: O(3*n) = O(n)
+# Space Complexity : O(m*n)
```

Comment:
here no. of column are 3 only. so space becomes O(n) where n is the no. of houses